### PR TITLE
Fullstack/alum approval

### PR DESCRIPTION
**Pull request by awilli70**
Modified alumniSchema to include verified field

New route on navbar - if user is verified, they will see the Verify Alumni tab

New route in alumni.js
GET alumni/unverified
Sends back all unverified alumni

For testing purposes, /alumni/ registered users are automatically verified, while seeded alumni aren't

Unverified:
Screen Shot 2020-05-27 at 12 28 22 AM

### CODE REVIEW

**Comment by awilli70** [comment]
This branch will most likely create merge conflicts upon completion of alumni preference setting 
It should not be merged until after preference setting is complete

By that point this will hopefully also be more feature complete but in its present state it is close to MVP

When a profile is verified it will send back a new list of entries - this will necessitate another route, `POST /alumni/verify/`

GOALS:
popup modals on profile click to display the profile

File: client/src/App.js [hunk 1]
```diff
@@ -139,6 +147,7 @@ class App extends Component {
     this.state = {
       loggedIn: false,
       fetchingAuth: true,
+      verified: Boolean,
```
Comment (reshadbinharun) [inline client/src/App.js:150]: Won't this typescript notation throw an error?

File: client/src/App.js [hunk 2]
```diff
@@ -170,6 +179,7 @@ class App extends Component {
         this.setState({
           role: role,
           userDetails: profile,
+          verified: profile.verified,
```
Comment (reshadbinharun) [inline client/src/App.js:182]: So apologies for being unclear about this.
`verified` would mean email verified. Once user's sign up they would get an email with a verification link they would have to click (to be implemented) to confirm their emails.
`approved` is the field we should be using!

File: client/src/App.js [hunk 3]
```diff
@@ -349,7 +374,7 @@ class App extends Component {
                   </> :
                   <Redirect to={"/login"}/>
               }
-          />
+          />
```
Comment (reshadbinharun) [inline client/src/App.js:377]: nit: white space

File: client/src/components/AlumniVerification.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+                }
+            ]
+        })
+    }
+    
+    handleDropdownChange = (e, { value }) => {
+        this.createDisplay(value)
+    }
+
+    handleClick(e) {
```
Comment (reshadbinharun) [inline client/src/components/AlumniVerification.js:91]: Is this for debugging?

Comment (reshadbinharun) [reply client/src/components/AlumniVerification.js:91]: May be we can name this `handleApproval`?

File: client/src/components/AlumniVerification.js [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        alert(e.currentTarget.dataset.id)
+    }
+
+
+    render(){
+        return (
+            <>
+            <Segment inverted color='red' secondary>Please only verify individuals that you know personally!</Segment>
```
Comment (reshadbinharun) [inline client/src/components/AlumniVerification.js:99]: Red seems a little error-state-y! We can talk more about this during standup!

File: server/routes/alumni.js [hunk 6]
```diff
@@ -41,7 +41,8 @@ router.post('/', async (req, res, next) => {
                 //posts: [{type: Schema.Types.ObjectId, ref: 'postSchema'}]
                 availabilities: availabilities,
                 timeZone: timeZone,
-                zoomLink: zoomLink
+                zoomLink: zoomLink,
+                verified: true
```
Comment (reshadbinharun) [inline server/routes/alumni.js:45]: I see, it kind of does make sense to move the `approved` from user collection to `Alumni` collection

Comment (reshadbinharun) [reply server/routes/alumni.js:45]: We can discuss this more during standup

File: client/src/App.js (hunk 1 above)
Comment (awilli70) [reply client/src/App.js:150]: It doesn’t, but I can set it to false by default after changing its name

**Comment by awilli70** [comment]
Mostly visual updates, but with some backend changes to reflect intended data models

Cards are now delivered in reverse order - this means that new profiles show up first, while older profiles that haven't / won't be approved will sink to the bottom

Warning message is now less aggressive:
Screen Shot 2020-05-27 at 5 47 40 PM

Clicking on a cards details will now pop up a profile modal to provide more information:
Screen Shot 2020-05-27 at 5 47 48 PM

I will update the warning message to show that this behavior is possible in future commits

**Comment by awilli70** [comment]
Finished!
POST /alumni/approve - approves an alumni by id
body:
id: ObjectId - returns all unapproved alumni and the name of the approved alumni

Visual changes and working verification:
Screen Shot 2020-05-27 at 10 10 58 PM
Screen Shot 2020-05-27 at 10 11 17 PM
Screen Shot 2020-05-27 at 10 11 36 PM

File: client/src/App.js [hunk 7]
```diff
@@ -272,14 +282,29 @@ class App extends Component {
...
                           activeItem={'requests'}
                       />
                       <div> Requests! </div>
                   </> :
                   <Redirect to={"/login"}/>
               }
           />
+          { this.state.userDetails.approved &&
+          <Route exact path={`/verify`} render={(props) => 
+                  this.state.loggedIn ?
```
Comment (reshadbinharun) [inline client/src/App.js:295]: nit: can we add parentheses/indentation to make the use of ternary more clear?

File: server/routes/alumni.js [hunk 8]
```diff
@@ -42,7 +42,8 @@ router.post('/', async (req, res, next) => {
                 //posts: [{type: Schema.Types.ObjectId, ref: 'postSchema'}]
                 availabilities: availabilities,
                 timeZone: timeZone,
-                zoomLink: zoomLink
+                zoomLink: zoomLink,
+                approved: true
```
Comment (reshadbinharun) [inline server/routes/alumni.js:46]: Let's set `approved` be false when alumni sign up

**Pull request merged**